clippy.toml

```diff
@@ -11,3 +11,6 @@ disallowed-methods = [
     { path = "crossbeam::channel::unbounded", reason = "Using an unbounded channel most likely will read to unbounded memory growth. Please use a bounded channel instead. If you are intentionally using unbounded channel, use `#[allow(clippy::disallowed_methods)]` to locally disable this check." },
     { path = "crossbeam_channel::unbounded", reason = "Using an unbounded channel most likely will read to unbounded memory growth. Please use a bounded channel instead. If you are intentionally using unbounded channel, use `#[allow(clippy::disallowed_methods)]` to locally disable this check." },
 ]
+disallowed-types = [
+    { path = "tokio::sync::Mutex", reason = "You should only use an asynchronous lock if you need to .await something while the lock is locked. Usually, this is not necessary, and you should avoid using an asynchronous lock when you can. Asynchronous locks are a lot slower than blocking locks. Please read how to share state effectively across async tasks https://tokio.rs/tokio/tutorial/shared-state. If you are intentionally using an async mutex, use `#[allow(clippy::disallowed_types)]` to locally disable this check." },
```

Comment:
Do you also want to forbid `tokio::sync::RwLock`? We could namely use `tokio::sync::RwLock` instead of `tokio::sync::Mutex` in PocketIC?

Reply:
I believe so. But I need to convince myself for this as well.